internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -268,6 +268,7 @@ func getTemplateData(ctx context.Context, rr *odhtypes.ReconciliationRequest) (m
 		"ApplicationNamespace": appNamespace,
 		"MetricsExporters":     make(map[string]string),
 		"MetricsExporterNames": []string{},
+		"MetricsTLSEnabled":    true,
```

Comment:
`MetricsTLSEnabled: true` is set here in `getTemplateData()` and again in `addMetricsData()` (line 408). This redundancy could lead to confusion. 

Remove the setting from one location. Since this is a metrics-specific setting, it's better to keep it only in `addMetricsData()` where it's documented.

```suggestion
        "MetricsTLSEnabled":    false, // Will be set to true in addMetricsData if metrics are configured
```

Or remove it entirely and only set it in `addMetricsData()`